persistence/migrations/migrations.go

```diff
@@ -15,9 +19,17 @@ func init() {
 
 // DiscoverSQLMigrations scans dir for migrations.
 func DiscoverSQLMigrations(dir string) error {
+	sqlDiscoveryMtx.Lock()
+	defer sqlDiscoveryMtx.Unlock()
+
+	if sqlDiscovered {
+		return nil
```

Comment:
I think this can be done in a more compact way using `sync.Once`

Reply:
I've updated it using `sync.Once`, it's quite a bit more compact. Thanks!